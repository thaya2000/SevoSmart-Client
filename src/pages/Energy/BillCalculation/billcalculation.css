
.account-details-form {
    max-width: 400px;
    margin: auto;
    padding: 20px;
    border-radius: 5px;
    margin-right:0;
    align-items: center; /* Center the content horizontally */
    justify-content: center; /* Center the content vertically */
   
  
  }

  .home{
  text-align: left;
  font-weight:200;
  font-family: 'Inter', sans-serif;
  font-size: 20px;
  }
  .order-summery {
    text-align: center;
  }
  
  .upper-container{
    max-width: 500px;
    padding: 40px;
    border-radius: 5px;
    margin-left:50px;
    align-items: center; /* Center the content horizontally */
    justify-content: center;
    margin-top: 0;
    /* border:4px solid; */
   
    
  }
  h2 {
    font-family: 'Inter', sans-serif; 
    font-weight: 600; /* Adjust the font weight (e.g., 600 for semi-bold) */
    font-size: 32px; /* Set the font size to 64 pixels */
    margin-bottom: 35px;
  }
  
  label {
    font-family: 'Inter', sans-serif; /* Use the 'Inter' font, fallback to a sans-serif font */
    display: block;
  
    margin-bottom: 15px;
  }
  
  .form-group {
    margin-bottom: 15px;
  }
  
  label {
    display: block;
    margin-bottom: 15px;
  }
  
  input {
    width: 100%;
    padding: 8px;
    box-sizing: border-box;
    border: 1px solid #ccc;
    border-radius: 3px;
    outline: none;
    margin-bottom: 15px;
    background-color:rgb(204, 203, 203); /* Set background color to grey */
  }
  
  button {
    width: 300px;
    padding: 10px;
    background-color: rgba(51, 75, 161, 1);
    color: #fff;
    border: none;
    border-radius: 25px;
    cursor: pointer;
    align-self: center;
    margin-top: 15px; 
    
  }
  
  .agreement-statement {
    margin-top: 15px; 
    text-align: left; 
    margin-bottom: 15px;
    color: rgb(0, 0, 0);

  }

  .agreement-statementA {
    margin-top: 15px; 
    text-align: left; 
    margin-bottom: 15px;
    color: rgb(0, 0, 0);

  }
  
  .agreement-statement p {
    margin: 0; 
    line-height: 1.5; 
    color: #555; 
  }
  
 

.img-container {
    display:flex;
    margin-left: 50px;
  }
  
  .image {
    width: 70%; 
    height: 25%;
    border-radius:30px;
    margin-top: 50px;
    margin-left: 70px; 
  }
  
  .power{
    text-align: center;
    color: rgb(0, 0, 0);
    font-weight: 600;
    margin-left: 30px;
    font-size:32px;
    font-family:'Inter',sans-serif;
  }

  .main-container {
    width: 50%;
  margin-right: 100px; 
   border:4px}

  .cproduct{

    font-weight: 400;
    font-family: 'Inter',sans-serif;
    font-size: 32px;
    margin-bottom: 20px; 
    
      
  }

  .help-text{
    margin-top: 20px;
    margin-bottom: 15px;
  }

  .product-image{
    border-radius: 25px;
    margin-top: 15px;
  }


  .product-name {
    text-align: left; 
    font-weight:600; 
    margin-top: 10px;
  }
  
  
  
 
  
  .newBill{
    margin-left: 200px;
    margin-top: 50px;
    font-family: 'Inter',sans-serif;
    font-size:30;
    font-weight: 600;
  }

  .squares-container {
    display: flex;
    
  }
  
  .square1 {
    width: 20px; 
    height: 20px; 
    background-color: rgb(31, 157, 157); 
    margin-left: 200px; 
    margin-top: 20px;
    display:inline-block;
    border-radius: 3px;
  }

  .square2 {
    width: 20px; 
    height: 20px; 
    background-color: rgb(166, 164, 164); 
    margin-left: 80px; 
    margin-top: 20px;
    display: inline-block;
    border-radius: 3px;
  }

  .text1 {
    display: inline-block;
    color: #151414; 
    margin-left: 20px; 
  }

  .text2{
    display: inline-block;
    color: #151414; 
    margin-left: 20px;
    margin-bottom: 40px; 
}
  .text3{
   
    color: #151414; 
    margin-left: 0px;
    margin-top: 10px;}
    

    .text4{
   
    color: #151414; 
    margin-left: 0px;
    margin-top: 10px;
        


  }
  .square3 {
    width: 100px; 
    height: 150px; 
    background-color: rgb(31, 157, 157); 
    margin-left: 0; 
    margin-top: 20px;
     border-radius: 10px;}

    .square4 {
    width: 100px; 
    height: 150px; 
    background: linear-gradient(to top,rgb(31, 157, 157)10%, #cccccc 10%, #ccc 100%); 
    margin-left: 0px; 
    margin-top: 20px;
   border-radius: 10px;
   transition: height 0.3s ease;
 
    }

    .container {
   display: flex;
   margin-left: 80px;
   
   
      }
    
       
         
   .A-container {
    /* border-color: #151414; */
    margin-top: 0px;
    width: 250px;
    display: flex;
    flex-direction: column;
    margin-left: 90px;
    margin-right: 10px;
    /* border-width: 4px; */
    overflow: hidden;
    padding:10px;
    

    
  }
  
  
  .B-container{
   
    /* border-width: 4px; */
    margin-top:0px;
    width:250px;
    display: flex;
    flex-direction: column;
    margin-left: 0px;
    padding:10px;
  }
  
   .D-container{

    margin-top:15px;
    margin-left:15px;
    margin-bottom: 15px;
    transition: background-color 0.3s ease;
  
   }

   

   .text5{
   
    color: #151414; 
    margin-left: 0px;
    margin-top: 20px;
    font-weight: 500;
    font-size: 20px;
   }

   .M-container {
    height:150px;
    display: flex;
    justify-content: center;
    /*border:solid 4px black;*/
    
    
       }
    .E-container {
        border: rgba(138, 139, 144, 0.811) solid 2px;
        margin-top: 20px;
        width:fit-content;
        padding:10px;
        margin-bottom: 20px;
    }

    .your-amount{
        margin-left:110px;
        font-weight: 600;
    }

    .text6{
        font-weight: 600;
        margin-left: 10px;
       
    }

    .text7{
        margin-left: 10px;
        
    }

    .text8{

        margin-top:30px;
        color: #151414; 
        margin-left: 0px;
        font-weight: 500;
        font-size: 18px;
    }
    
    .c-container {
        /* Other styles for your container */
      
        /* Add a smooth transition effect */
        transition: transform 0.5s ease;
        justify-content: center;
      }
      
      .c-container:hover {
        transform: scale(1.1); 
      }
          
     

    .A-container:hover .square3 {
    transform: scale(1.1); /* Increase the size on hover */
    }
  
    .B-container:hover .square4 {
    transform: scale(1.1); /* Increase the size on hover */
    }
  
 
   .square3,
   .square4 {
    padding-bottom: 10px;
    transition: transform 0.5s ease;
   }
  
 
    .billhead{
    margin-left: 180px;
    margin-top: 50px;
    font-family: 'Inter',sans-serif;
    font-size:30px;
    font-weight: 600;
    color: rgb(22, 126, 126);
    margin-bottom: 20px;

    }    


    .bullet-points {
      margin-top: 20px;
      margin-left: 20px;
      line-height: 1;
      list-style-type: disc; /* You can change 'disc' to 'circle' or 'square' if you prefer different bullet styles */
      /* Add any other styles you want for the list */
    }
    
    /* Additional styling for list items if needed */
    .bullet-points li {
      margin-bottom: 10px; /* Adjust as needed */
    }
    .energy-order-buttonA {
      display: flex;
      justify-content: center;
      width: 50%;
      height: 40px;
      padding: 10px;
      background-color: rgba(5, 105, 255, 1);
      color: #fff;
      border: none;
      border-radius: 25px;
      cursor: pointer;
      align-self: center;
      margin-left: 100px;
      margin-top: 25px; 
      font-size: 14px;
      transition: background-color 0.3s; /* Add a smooth transition effect for the background color */
    }
    
    .energy-order-buttonA:hover {
      background-color: rgba(111, 112, 113, 0.8); /* Change the background color on hover */
    }
    