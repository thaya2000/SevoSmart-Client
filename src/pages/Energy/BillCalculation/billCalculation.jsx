import React, { useState, useEffect } from "react";
// import Link  from 'react-router-dom';
import "./billcalculation.css";
import Image7 from "../../../Images/S7.jpg";
import { NavLink } from "react-router-dom";

function BillCalculation() {
  const [formData, setFormData] = useState({});

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // console.log("Combined Form submitted:", formData);
  };

  return (
    <div>
      <div className="img-container">
        <div>
          <div className="billhead">Average bill of your home</div>

          <div className="square1"></div>
          <span className="text1">Your electric bill</span>
          <div className="square2"></div>
          <span className="text2">Solar payment</span>
          <div className="container">
            <div className="A-container">
              <span
                className="text3"
                style={{ display: "flex", alignItems: "center" }}
              >
                15000 lkr /month <br />
                without solar
              </span>
              <div className="square3"></div>
            </div>
            <div className="B-container">
              <span
                className="text4"
                style={{ display: "flex", alignItems: "center" }}
              >
                1500 lkr /month <br />
                with solar
              </span>
              <div className="square4"></div>
            </div>
          </div>
        </div>
        <div className="main-container">
          <div className="upper-container">
            {/* Content for the container above "Enter Account Details" */}
            <div className="power">Your Home</div>
            <div className="agreement-statementA">
              Produce clean energy with the lowest cost solar panels, <br />
              price match guaranteed
            </div>

            <div className="M-container">
              <div className="c-container">
                <img
                  src={Image7}
                  alt="3 Kw Solar System Off Grid"
                  className="product-image"
                  style={{ width: "150px", height: "auto" }}
                />
              </div>

              <div className="D-container">
                <span className="agreement-statement">
                  1500 Homes in Trincomalee
                  <br />
                  have solar
                  <br />
                  100/2, Green Road ,<br />
                  Trincomalee, SriLanka.
                </span>
              </div>
            </div>
            <div>
              <span className="text5">
                Solar Panels <br /> produce Green Energy
              </span>
            </div>

            <div className="E-container">
              <span className="text6">Meet current needs</span>
              <span className="your-amount">2,500,000 lkr</span>
              <br />
              <span className="text7">4 kW solar</span>

              <ul className="bullet-points">
                <li>6,111 kWh /yr Solar Production</li>
                <li>125% Energy Offset</li>
              </ul>
            </div>

            <div>
              <span className="text8">Order your system</span>
            </div>

            <NavLink to="/energy">
              <div className="energy-order-buttonA">continue to order</div>
            </NavLink>
          </div>
        </div>
      </div>
    </div>
  );
}

export default BillCalculation;
