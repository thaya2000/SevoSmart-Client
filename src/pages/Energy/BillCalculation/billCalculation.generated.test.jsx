import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import BillCalculation from './BillCalculation';

// Mocks for CSS and images
vi.mock('./billcalculation.css');
vi.mock('../../../Images/S7.jpg');
vi.mock('react-router-dom', () => ({
  NavLink: vi.fn().mockImplementation(({ children }) => <div>{children}</div>) // Mock NavLink as a simple div
}));

describe('<BillCalculation>', () => {
  it('should render component', () => {
    const { asFragment } = render(<BillCalculation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
