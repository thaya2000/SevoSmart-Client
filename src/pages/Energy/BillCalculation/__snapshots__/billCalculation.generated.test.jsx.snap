// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<BillCalculation> > should render component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="img-container"
    >
      <div>
        <div
          class="billhead"
        >
          Average bill of your home
        </div>
        <div
          class="square1"
        />
        <span
          class="text1"
        >
          Your electric bill
        </span>
        <div
          class="square2"
        />
        <span
          class="text2"
        >
          Solar payment
        </span>
        <div
          class="container"
        >
          <div
            class="A-container"
          >
            <span
              class="text3"
              style="display: flex; align-items: center;"
            >
              15000 lkr /month 
              <br />
              without solar
            </span>
            <div
              class="square3"
            />
          </div>
          <div
            class="B-container"
          >
            <span
              class="text4"
              style="display: flex; align-items: center;"
            >
              1500 lkr /month 
              <br />
              with solar
            </span>
            <div
              class="square4"
            />
          </div>
        </div>
      </div>
      <div
        class="main-container"
      >
        <div
          class="upper-container"
        >
          <div
            class="power"
          >
            Your Home
          </div>
          <div
            class="agreement-statementA"
          >
            Produce clean energy with the lowest cost solar panels, 
            <br />
            price match guaranteed
          </div>
          <div
            class="M-container"
          >
            <div
              class="c-container"
            >
              <img
                alt="3 Kw Solar System Off Grid"
                class="product-image"
                src="/src/Images/S7.jpg"
                style="width: 150px; height: auto;"
              />
            </div>
            <div
              class="D-container"
            >
              <span
                class="agreement-statement"
              >
                1500 Homes in Trincomalee
                <br />
                have solar
                <br />
                100/2, Green Road ,
                <br />
                Trincomalee, SriLanka.
              </span>
            </div>
          </div>
          <div>
            <span
              class="text5"
            >
              Solar Panels 
              <br />
               produce Green Energy
            </span>
          </div>
          <div
            class="E-container"
          >
            <span
              class="text6"
            >
              Meet current needs
            </span>
            <span
              class="your-amount"
            >
              2,500,000 lkr
            </span>
            <br />
            <span
              class="text7"
            >
              4 kW solar
            </span>
            <ul
              class="bullet-points"
            >
              <li>
                6,111 kWh /yr Solar Production
              </li>
              <li>
                125% Energy Offset
              </li>
            </ul>
          </div>
          <div>
            <span
              class="text8"
            >
              Order your system
            </span>
          </div>
          <div>
            <div
              class="energy-order-buttonA"
            >
              continue to order
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
